core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-add-variable/store/dot-add-variable.store.ts

```diff
@@ -58,4 +86,69 @@ export class DotAddVariableStore extends ComponentStore<DotAddVariableState> {
             })
         );
     });
+
+    // You can add here a new fieldType and add the fields that it has
+    private readonly fields: Record<
+        string,
+        (variableContent: DotFieldContent) => DotFieldContent[]
+    > = {
+        Image: ({ variable, name, fieldTypeLabel }) => [
+            {
+                name: `${name}: ${this.dotMessage.get('Image-Identifier')}`,
+                variable: `${variable}ImageIdentifier`,
+                codeTemplate: this.getCodeTemplate.default(`${variable}ImageIdentifier)`),
+                fieldTypeLabel
+            },
+            {
+                name: `${name}: ${this.dotMessage.get('Image')}`,
+                variable: `${variable}Image`,
+                codeTemplate: this.getCodeTemplate.image(variable),
+                fieldTypeLabel
+            },
+            {
+                name: `${name}: ${this.dotMessage.get('Image-Title')}`,
+                variable: `${variable}ImageTitle`,
+                codeTemplate: this.getCodeTemplate.default(`${variable}ImageTitle)`),
+                fieldTypeLabel
+            },
+            {
+                name: `${name}: ${this.dotMessage.get('Image-Extension')}`,
+                variable: `${variable}ImageExtension`,
+                codeTemplate: this.getCodeTemplate.default(`${variable}ImageExtension)`),
+                fieldTypeLabel
+            },
+            {
+                name: `${name}: ${this.dotMessage.get('Image-Width')}`,
+                variable: `${variable}ImageWidth`,
+                codeTemplate: this.getCodeTemplate.default(`${variable}ImageWidth)`),
+                fieldTypeLabel
+            },
+            {
+                name: `${name}: ${this.dotMessage.get('Image-Height')}`,
+                variable: `${variable}ImageHeight`,
+                codeTemplate: this.getCodeTemplate.default(`${variable}ImageHeight)`),
+                fieldTypeLabel
+            }
+        ],
+        default: ({
+            variable,
+            name,
+            fieldTypeLabel,
+            codeTemplate = this.getCodeTemplate.default(variable)
+        }) => [
+            {
+                name,
+                variable,
+                fieldTypeLabel,
+                codeTemplate
+            }
+        ]
+    };
+
+    // You can add here a new variable and add the custom code that it has
+    private readonly getCodeTemplate: Record<string, (variable: string) => string> = {
+        image: (variable) =>
+            `#if ($UtilMethods.isSet(\${${variable}ImageURI}))\n    <img src="$!{dotContentMap.${variable}ImageURI}" alt="$!{dotContentMap.${variable}ImageTitle}" />\n#end`,
+        default: (variable) => `$!{dotContentMap.${variable}}`
+    };
```

Comment:
I like maps

Reply:
I tried to do it in a way that we don't have to add too much code when we have to implement the rest of the custom code ;)